catalogs/catalog-lakehouse-iceberg/build.gradle.kts

```diff
@@ -68,11 +68,11 @@ dependencies {
   testImplementation(project(":server"))
   testImplementation(project(":server-common"))
 
-  implementation(libs.hadoop2.common) {
+  implementation(libs.hadoop3.common) {
     exclude("com.github.spotbugs")
   }
-  implementation(libs.hadoop2.hdfs)
-  implementation(libs.hadoop2.mapreduce.client.core)
+  implementation(libs.hadoop3.hdfs)
+  implementation(libs.hadoop3.mapreduce.client.core)
```

Comment:
@FANNG1 can you please explain why and where do we need Hadoop for Iceberg?

Reply:
We need pass HDFS configuration to init HiveCatalog
```java
  private static HiveCatalog loadHiveCatalog(Map<String, String> properties) {
    HiveCatalog hiveCatalog = new HiveCatalog();
    HdfsConfiguration hdfsConfiguration = new HdfsConfiguration();
    properties.forEach(hdfsConfiguration::set);
    hiveCatalog.setConf(hdfsConfiguration);
    hiveCatalog.initialize("hive", properties);
    return hiveCatalog;
  }
```

Reply:
When using the Iceberg rest catalog, the rest catalog server will write the metadata file. https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java#L345

Reply:
I see, thanks @FANNG1 @coolderli . I would suggest to upgrade to hadoop 3 only in `catalog-hadoop`. We need more investigation and works to support Hadoop 3 in all for Gravitino.